src/MainnetController.sol

```diff
@@ -536,6 +586,42 @@ contract MainnetController is AccessControl {
             maxSlippage        : maxSlippages[pool]
         }));
     }
+    
+    /**********************************************************************************************/
+    /*** Relayer UniswapV3 functions                                                            ***/
+    /**********************************************************************************************/
+    function swapUniswapV3(
+        address pool,
+        address tokenIn,
+        uint256 amountIn,
+        uint256 minAmountOut,
+        uint24  swapMaxTickDelta,
+        uint256 deadline
+    )
+        external returns (uint256 amountOut)
+    {
+        _checkRole(RELAYER);
+        require(uniswapV3Router != address(0), "MainnetController/router-not-set");
+
+        amountOut = UniswapV3Lib.swap(
+            UniswapV3Lib.UniV3Context({
+                proxy       : proxy,
+                rateLimits  : rateLimits,
+                rateLimitId : LIMIT_UNISWAP_V3_SWAP,
+                pool        : pool,
+                deadline    : deadline
```

Comment:
Why is deadline part of the context and not part of the swap params?

Reply:
deadline is a common variable among swap/addLiquidity/removeLiquidity